### PR TITLE
Adding new AvoidLookAtSphere task map.

### PR DESCRIPTION
**Pull request by cmower**
* Avoids "looking at" a number of spherical objects.
* Assumes all objects postions/radii are known.
* Can be used as either a cost term or an inequality constraint.
* I was having some strange issues with git on branch for #581 and thought it was easier to make a new branch. 

# Checklist

- [x] code formatting: run `find -name '*.cpp' -o -name '*.h' -o -name '*.hpp' | xargs clang-format-3.9 -style=file -i` in the root directory

- [x] add unit test(s)

- [x] ensure tests build and check results: run `catkin run_tests exotica && catkin_test_results`

### CODE REVIEW

File: exotations/exotica_core_task_maps/src/avoid_look_at_sphere.cpp [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    for (int i = 0; i < n_objects_; ++i)
+    {
+        const double frac = Eigen::Map<Eigen::Vector3d>(kinematics[0].Phi(i).p.data).topRows<2>().squaredNorm() / radii_squared_(i);
+        if (frac < 1.0) phi(i) = 1.0 - 2.0 * frac + frac * frac;
+    }
+}
+
+void AvoidLookAtSphere::UpdateAsInequalityConstraintWithoutJacobian(Eigen::VectorXdRefConst x, Eigen::VectorXdRef phi)
+{
+    for (int i = 0; i < n_objects_; ++i) phi(i) = radii_squared_(i) - Eigen::Map<Eigen::Vector3d>(kinematics[0].Phi(i).p.data).topRows<2>().squaredNorm();
```
Comment (VladimirIvan) [inline exotations/exotica_core_task_maps/src/avoid_look_at_sphere.cpp]: Use `Eigen::Map<Eigen::Vector2d>(kinematics[0].Phi(i).p.data)` instead of `Eigen::Map<Eigen::Vector3d>(kinematics[0].Phi(i).p.data).topRows<2>()`.

**Comment by wxmerkt** [comment]
What are the changes between this and #581? Closing old PRs and opening new ones that are semi-identical duplicates review work and makes it hard what changed :-/

**Comment by cmower** [comment]
Sorry, I wouldn't usually do this but I just thought it would be easier to create new branch and pr rather than work out what I'd done wrong with git. 

The only changes were to remove the constructors/destructors in header and source file. Will address @VladimirIvan point above asap and push.

File: exotations/exotica_core_task_maps/src/avoid_look_at_sphere.cpp [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    }
+    else
+    {
+        UpdateTaskMapWithoutJacobian = &AvoidLookAtSphere::UpdateAsInequalityConstraintWithoutJacobian;
+        UpdateTaskMapWithJacobian = &AvoidLookAtSphere::UpdateAsInequalityConstraintWithJacobian;
+    }
+
+    if (debug_ && Server::IsRos())
+    {
+        pub_markers_ = Server::Advertise<visualization_msgs::MarkerArray>("alas_objs", 1, true);
```
Comment (wxmerkt) [inline exotations/exotica_core_task_maps/src/avoid_look_at_sphere.cpp]: Please change the name of the topic, `alas_objs` is quite hard to parse. `avoid_look_at_sphere_objects` is still fine.

File: exotations/exotica_core_task_maps/src/avoid_look_at_sphere.cpp [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+REGISTER_TASKMAP_TYPE("AvoidLookAtSphere", exotica::AvoidLookAtSphere);
+
+namespace exotica
+{
+void AvoidLookAtSphere::UpdateAsCostWithoutJacobian(Eigen::VectorXdRefConst x, Eigen::VectorXdRef phi)
+{
+    for (int i = 0; i < n_objects_; ++i)
+    {
+        const double frac = Eigen::Map<Eigen::Vector2d>(kinematics[0].Phi(i).p.data).squaredNorm() / radii_squared_(i);
+        if (frac < 1.0) phi(i) = 1.0 - 2.0 * frac + frac * frac;
```
Comment (wxmerkt) [inline exotations/exotica_core_task_maps/src/avoid_look_at_sphere.cpp]: What happens to `phi` if `frac >= 1.0`? That's undefined, perhaps set it explicitly to 0?

Comment (cmower) [reply exotations/exotica_core_task_maps/src/avoid_look_at_sphere.cpp]: Will explicitly set to zero.
